cabal-install/Distribution/Client/DistDirLayout.hs

```diff
@@ -107,14 +111,20 @@ data DistDirLayout = DistDirLayout {
      }
 
 
+data StoreDirLayout = StoreDirLayout {
```

Comment:
Whatever we end up doing, we should have a source code note or ticket mentioning this so we can solve this eventually.

The strategy suggested here is similar to what chak described at https://ghc.haskell.org/trac/ghc/ticket/11587#comment:4 (which you have commented in.) Probably the biggest obstacle is this: according to rwbarton, it's not possible to store subdir names in the NEEDED entries if the SONAME is present but does not encode the subdir. darchon points out some other problems at https://ghc.haskell.org/trac/ghc/ticket/12479#comment:24 and it seems that in the ensuing discussion, it was agreed that it is better to go a different route. Also CC'ing @christiaanb